cli/src/main/java/org/jboss/as/cli/handlers/DeploymentHandler.java

```diff
@@ -64,25 +64,29 @@ protected void listDeployments(CommandContext ctx, boolean l) throws CommandForm
         }
 
         final StrictSizeTable table = new StrictSizeTable(names.size());
-        final List<Property> descriptions = getDeploymentDescriptions(ctx, names).asPropertyList();
-        for(Property prop : descriptions) {
-            final ModelNode step = prop.getValue();
-            if(step.hasDefined(Util.RESULT)) {
-                final ModelNode result = step.get(Util.RESULT);
-                table.addCell(Util.NAME, result.get(Util.NAME).asString());
-                table.addCell(Util.RUNTIME_NAME, result.get(Util.RUNTIME_NAME).asString());
-                if(result.has(Util.ENABLED)) {
-                    table.addCell(Util.ENABLED, result.get(Util.ENABLED).asString());
+        try {
+            final List<Property> descriptions = getDeploymentDescriptions(ctx, names).asPropertyList();
+            for(Property prop : descriptions) {
+                final ModelNode step = prop.getValue();
+                if(step.hasDefined(Util.RESULT)) {
+                    final ModelNode result = step.get(Util.RESULT);
+                    table.addCell(Util.NAME, result.get(Util.NAME).asString());
+                    table.addCell(Util.RUNTIME_NAME, result.get(Util.RUNTIME_NAME).asString());
+                    if(result.has(Util.ENABLED)) {
+                        table.addCell(Util.ENABLED, result.get(Util.ENABLED).asString());
+                    }
+                    if(result.has(Util.STATUS)) {
+                        table.addCell(Util.STATUS, result.get(Util.STATUS).asString());
+                    }
                 }
-                if(result.has(Util.STATUS)) {
-                    table.addCell(Util.STATUS, result.get(Util.STATUS).asString());
+                if(!table.isAtLastRow()) {
+                    table.nextRow();
                 }
             }
-            if(!table.isAtLastRow()) {
-                table.nextRow();
-            }
+            ctx.printLine(table.toString());
+        } catch (Exception e) {
```

Comment:
In the existing code, the CommandFormatException is not thrown because of an Exception being thrown. It is thrown at the end of the block.
If a RuntimeException or CommandFormatException is thrown, it will be thrown, no need to re-throw it. So yes, replacing the throw with printLine is safe.